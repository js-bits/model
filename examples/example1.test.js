// eslint-disable-next-line import/no-extraneous-dependencies
import { jest } from '@jest/globals';

describe('Examples', () => {
  beforeEach(() => {
    global.console = { log: jest.fn() };
  });
  afterEach(() => {
    jest.resetModules();
  });

  test('Example 1', done => {
    import('./example1.js').then(() => {
      expect(console.log).toHaveBeenCalledTimes(2);
      expect(console.log.mock.calls[0]).toEqual(['[object Model]']);
      expect(console.log.mock.calls[1]).toEqual([
        JSON.stringify(
          {
            firstName: 'Trygve',
            lastName: 'Reenskaug',
            yearBorn: 1930,
            verified: null,
          },
          null,
          '  '
        ),
      ]);
      done();
    });
  });
});
