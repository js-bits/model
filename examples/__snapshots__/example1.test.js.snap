// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examples Example 1 1`] = `
Array [
  Array [
    "[object Model]",
  ],
  Array [
    "{
  \\"firstName\\": \\"Trygve\\",
  \\"lastName\\": \\"Reenskaug\\",
  \\"yearBorn\\": 1930
}",
  ],
]
`;
